cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/sparksql/CellIterator.java

```diff
@@ -0,0 +1,386 @@
+                    // regular columns.
+                    next = new Cell(values, firstProjectedValueColumnPositionOrZero, newRow, rowData.getTimestamp());
+                    return true;
+                }
+
+                continue;
+            }
+
+            CqlField field = cqlTable.getField(columnName);
+            if (field == null)
+            {
+                LOGGER.warn("Ignoring unknown column columnName='{}'", columnName);
+                continue;
+            }
+
+            // Deserialize value field or static column and update 'values' Object[] array
+            deserializeField(field);
+
+            // Static column, so continue reading entire CQL row before returning
+            if (field.isStaticColumn())
+            {
+                continue;
+            }
+
+            // Update next Cell
+            next = new Cell(values, field.position(), newRow, rowData.getTimestamp());
+
+            return true;
+        }
+
+        // Finished so close
+        next = null;
+        close();
+        return false;
+    }
+
+    protected String decodeString(@Nullable ByteBuffer buffer) throws CharacterCodingException
+    {
+        return buffer == null ? null : ByteBufferUtils.string(buffer);
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        if (!closed)
+        {
+            scanner.close();
+            closed = true;
+            long runtimeNanos = System.nanoTime() - startTimeNanos;
+            LOGGER.info("Closed CompactionScanner runtimeNanos={}", runtimeNanos);
+            stats.closedSparkCellIterator(runtimeNanos);
+        }
+    }
+
+    /* Iterator Helpers */
+
+    /**
+     * If it is a new partition see if we can skip (e.g. if partition outside Spark worker token range), otherwise re-build partition keys
+     */
+    private void maybeRebuildPartition()
+    {
+        if (!rowData.isNewPartition())
+        {
+            return;
+        }
+
+        // Skip partitions not in the token range for this Spark partition
+        newRow = true;
+
+        for (CqlField field : cqlTable.staticColumns())
+        {
+            // We need to reset static columns between partitions, if a static column is null/not-populated
+            // in the next partition, then the previous value might be carried across
+            values[field.position()] = null;
+        }
+
+        skipPartition = !isInPartition(partitionId, rowData.getToken(), rowData.getPartitionKey());
+        if (skipPartition)
+        {
+            stats.skippedPartitionInIterator(rowData.getPartitionKey(), rowData.getToken());
+            return;
+        }
+
+        // Or new partition, so deserialize partition keys and update 'values' array
+        readPartitionKey(typeConverter, rowData.getPartitionKey(), cqlTable, this.values, stats);
+    }
+
+    public abstract boolean isInPartition(int partitionId, BigInteger token, ByteBuffer partitionKey);
```

Comment:
Can you group the abstract methods in this file together and add java doc for them?